web/src/settings_components.ts

```diff
@@ -1083,6 +1086,10 @@ export function populate_data_for_realm_settings_request(
     return data;
 }
 
+const is_stream_permission_group_setting = (setting_name: string): boolean =>
+    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+    stream_permission_group_settings.includes(setting_name as stream_permission_group_setting_type);
```

Comment:
Think we can parse `setting_name` through a zod enum schema instead of using `as stream_permission_group_setting_type`. We have a similar `group_setting_name_schema` in the same file.